lib/cartopy/_crs.pyx

```diff
@@ -446,15 +436,21 @@ cdef class CRS:
 
         # call proj. The result array is modified in place. This is only
         # safe if npts is not 0.
-        if npts:
-            status = _safe_pj_transform(src_crs, self, npts, 3,
-                                        result[:, 0], result[:, 1],
-                                        result[:, 2])
+        if npts == 0:
+            return result
+        status = _safe_pj_transform(src_crs, self, npts, 3,
+                                    result[:, 0], result[:, 1],
+                                    result[:, 2])
+
+        if trap and status == -14 or status == -20:
```

Comment:
So I'm fairly certain this was wrong before, but the expression doesn't make sense here. It should be 
```suggestion
        if trap and status in (-14, -20):
```